infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -343,6 +376,65 @@ public void prefixTrim(long address) {
         logMetadata.prefixTrim(address);
     }
 
+    @Override
+    public void suffixTrim() {
+
+        // Have a global tail copy is necessary, since degrade to reset is dependent on it
+        long globalTailCopy = logMetadata.getGlobalTail();
+
+        try {
+            long committedTail = dataStore.getCommittedTail();
+
+            // Check if committed tail is not less than starting address
+            if (isTrimmed(committedTail)) {
+                throw new SuffixTrimException("local committed tail is trimmed",
+                        committedTail, globalTailCopy);
+            }
+
+            // Check if committed tail is greater than global tail
+            if (committedTail > globalTailCopy) {
+                throw new SuffixTrimException("local committed tail is greater than global tail",
+                        committedTail, globalTailCopy);
+            }
+
+            // Close all segment handlers and delete suffix log files
+            long committedSegment = committedTail / RECORDS_PER_LOG_FILE;
```

Comment:
I know you did not start it but I think this segment calculation should be put into a separate function.